include/mpm_base.tcc

```diff
@@ -368,6 +368,15 @@ bool mpm::MPMBase<Tdim>::initialise_particles() {
         throw std::runtime_error("Particles volumes are not properly assigned");
     }
 
+    // Default particle porosity as 0
+    double porosity = 0;
+    // Read and assign porosity (volume fraction and phase volume)
+    if (!particle_props.at("porosity").empty())
```

Comment:
1. Move this function to two phase solver.

2. `porosity` is a double value, so let's not use `.empty()`.
Change this to 
```
if (particle_props.find("porosity") != particle_props.end())
   porosity = particle_props["porosity"].template get<double>();
```